lib/src/Pos/Logic/Types.hs

```diff
@@ -53,6 +54,9 @@ data Logic m = Logic
       -- be a tip, whereas trying to get a block that isn't in the database is
       -- normal.
     , getTip             :: m (Either GetTipError Block)
+      -- TBD useful to have this and getTip? There are existing different
+      -- implementations so presumably yes?
+    , getTipHeader       :: m (Either GetTipError BlockHeader)
```

Comment:
Oh, wait a minute, it violates exception handling guidelines. Potentially impure code [must not](https://github.com/input-output-hk/cardano-sl/blob/6e1da775afdc5e59f13daadda14c674ab29fca53/docs/exceptions.md#potentially-impure-code-regular-errors)  return `m (Either e a)` when `e` has `Exception` instance.

The same is true for some other methods from `Logic`. I know it's already in `develop`, but it should be fixed.

Reply:
So if I get rid of the exception instance, I can keep it as is?

Reply:
Ok, I suppose that for `getTipHeader`, there really is no "normal" error. Contrast with `getBlockHeaders`, where having a `Left` option is useful to describe completely normal cases like an inconsistent checkpoint list and tip header (maybe tip is actually behind the checkpoints).